client/my-sites/plans-features-main/index.tsx

```diff
@@ -216,6 +218,9 @@ const PlansFeaturesMain = ( {
 	const [ masterbarHeight, setMasterbarHeight ] = useState( 0 );
 	const translate = useTranslate();
 	const plansComparisonGridRef = useRef< HTMLDivElement >( null );
+	const storageAddOns = useAddOns( siteId as number ).filter(
```

Comment:
Using `as` for type casting here is dangerous as it turns off type checking for this variable. If the variable or the function signature were ever to change, TypeScript would be unable to prevent fatal errors from happening.

In this case, the variable's type is `number | null`, and `useAddOns` accepts `number | undefined`. So I think all you need to do is either change `useAddOns` to also accept null or write this as:

```suggestion
	const storageAddOns = useAddOns( siteId ?? undefined ).filter(
```

Reply:
Committed in https://github.com/Automattic/wp-calypso/pull/81338/commits/7f500b3d90e165a801e7eea85a51c6047ccde17f

Thanks for the suggestion @sirbrillig 🙏